### PR TITLE
Milestone 1: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by jaflih**
### Milestone 1: add Person, Student and Teacher classes

- Created class Person with the following:
    - Instance vars: @id, @name, and @age.
    - Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
    - Getters for @id, @name, and @age.
    - Setters for @name and @age.
    - Private method is_of_age? that returns true if @age is greater or equal to 18 and false otherwise.
    - Public method can_use_services? that returns true if person is of age or if they have permission from parents.
    
- Created class Student with the following:
    - Inherits from Person.
    - Constructor extends parent's constructor by adding @classroom and a parameter for it.
    - Method play_hooky that returns "¯\(ツ)/¯".
    
- Created class Teacher with the following:
    - Inherits from Person.
    - Constructor extends parent's constructor by adding @specialization and a parameter for it.
    - Override can_use_services? so it always returns true.
    
- Each class is saved in a separate file.

### CODE REVIEW

File: lib/person.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (mikethreels) [inline lib/person.rb:9]: - [ ] According to the requirements there should be an instance variable setup for `@id` as well, you can set this up as a random number.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Jihane, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
### Because there where no changes made to the Readme file in this PR I'm not able to leave an inline comment, therefore I'll leave one here.
- [ ] The link to the MIT license is not working please have another look at [ the template](https://github.com/microverseinc/readme-template) for the readme file provided by Microverse and make sure the link to the MIT license works the same as in that template.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/person.rb (hunk 1 above)
Comment (jaflih) [reply lib/person.rb]: @mikethreels , ok it's done.
Thank you for your feedback

File: lib/student.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(age, name, parent_permission, classroom)
+    @classroom = classroom
+    super(age, name, parent_permission: parent_permission)
+  end
```
[the remark this answers is not in the document]
Comment (jaflih) [reply lib/student.rb]: Ok, it's done.

File: lib/teacher.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, name, parent_permission, specialization)
+    @specialization = specialization
+    super(age, name, parent_permission: parent_permission)
+  end
```
[the remark this answers is not in the document]
Comment (jaflih) [reply lib/teacher.rb]: Ok, it's done.

**Review by mikethreels — APPROVED** [review]
Hi Jihane, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jaflih** [comment]
@mikethreels 
thanks

**Pull request merged**